example/gluon/image_classification.py

```diff
@@ -172,16 +223,47 @@ def train(epochs, ctx):
         out = net(data)
         softmax = mx.sym.SoftmaxOutput(out, name='softmax')
         mod = mx.mod.Module(softmax, context=[mx.gpu(i) for i in range(num_gpus)] if num_gpus > 0 else [mx.cpu()])
+
+        # load model
+        if 'arg_params' in kwargs and 'aux_params' in kwargs:
+            arg_params = kwargs['arg_params']
+            aux_params = kwargs['aux_params']
+        else:
+            sym, arg_params, aux_params = _load_model(kv.rank)
+            if sym is not None:
+                assert sym.tojson() == softmax.tojson()
+
+        # save model
+        checkpoint = _save_model(kv.rank)
+
+        # learning rate
+        lr, lr_scheduler = _get_lr_scheduler()
+        optimizer_params = {
+            'learning_rate': lr,
+            'wd': opt.wd,
+            'lr_scheduler': lr_scheduler}
+        # Add 'multi_precision' parameter only for SGD optimizer
+        if opt.optimizer == 'sgd':
+            optimizer_params['multi_precision'] = True
+
+        # Only a limited number of optimizers have 'momentum' property
+        has_momentum = {'sgd', 'dcasgd', 'nag'}
+        if opt.optimizer in has_momentum:
+            optimizer_params['momentum'] = opt.momentum
+
         mod.fit(train_data,
+                begin_epoch=opt.load_epoch if opt.load_epoch else 0,
                 eval_data = val_data,
                 num_epoch=opt.epochs,
                 kvstore=opt.kvstore,
```

Comment:
I think `kvstore` here should be `kv` you created before, not the string.

Reply:
yes, I think this is a good suggestion, for it save a instance of 'kv', thx